### PR TITLE
Allow legacy targets to be used when manually specified

### PR DESCRIPTION
**Pull request by shadaj**

### CODE REVIEW

File: packages/apollo-cli/src/commands/codegen/generate.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ export default class Generate extends Command {
 
     let inferredTarget: TargetType = "" as TargetType;
     if (flags.target) {
-      if (["swift", "typescript", "flow", "scala"].includes(flags.target)) {
+      if (["swift", "typescript", "ts", "flow", "scala", "typescript-legacy", "ts-legacy", "flow-legacy"].includes(flags.target)) {
```
Comment (martijnwalraven) [inline packages/apollo-cli/src/commands/codegen/generate.ts]: Maybe we should get rid of the `ts` aliases? It seems confusing to have multiple names for the same target.

Comment (shadaj) [reply packages/apollo-cli/src/commands/codegen/generate.ts]: I was mostly thinking about compatibility with `apollo-codegen` here. Is it okay to drop target names now?

Comment (martijnwalraven) [reply packages/apollo-cli/src/commands/codegen/generate.ts]: As this is a new CLI, I think we should favor clarity over backwards compatibility. Plus, I think in `apollo-codegen` we didn't have `typescript-legacy` but `typescript-modern` so it's breaking anyway?

Comment (shadaj) [reply packages/apollo-cli/src/commands/codegen/generate.ts]: Good point, I'll drop `ts`.

**Pull request merged**